src/components/CouponDetails/helpers.test.jsx

```diff
@@ -0,0 +1,162 @@
+/* eslint-disable max-len */
+/* eslint-disable object-curly-newline */
+import {
+  SINGLE_USE, ONCE_PER_CUSTOMER, MULTI_USE, MULTI_USE_PER_CUSTOMER,
+} from '../../data/constants/coupons';
+import { ACTION_LABELS, ACTION_TYPES, COUPON_FILTER_TYPES, FILTER_OPTIONS } from './constants';
+import { getBASelectOptions, getFilterOptions, getFirstNonDisabledOption, shouldShowSelectAllStatusAlert } from './helpers';
+
+describe('getFilterOptions', () => {
+  test.each([
+    [MULTI_USE],
+    [MULTI_USE_PER_CUSTOMER],
+  ])('includes partially redeemed for partially redeemable coupons %p', (usageLimitation) => {
+    expect(getFilterOptions(usageLimitation)).toEqual(FILTER_OPTIONS);
+  });
+  test.each([
+    [SINGLE_USE],
+    [ONCE_PER_CUSTOMER],
+  ])('does not include partially redeemed for single-use coupons %p', (usageLimitation) => {
+    const expected = [{
+      label: 'Unassigned',
+      value: 'unassigned',
+    }, {
+      label: 'Unredeemed',
+      value: 'unredeemed',
+    }, {
+      label: 'Redeemed',
+      value: 'redeemed',
+    }];
+    expect(getFilterOptions(usageLimitation)).toEqual(expected);
+  });
+});
+
+describe('getFirstNonDisabledOption', () => {
+  test.each([
+    [[{ value: 'foo', disabled: true }, { value: 'bar' }], 'bar'],
+    [[{ value: 'foo' }, { value: 'bar', disabled: false }, { value: 'baz', disabled: true }], 'foo'],
+    [[{ value: 'foo', disabled: true }, { value: 'bar', disabled: true }, { value: 'baz', disabled: false }], 'baz'],
+  ])('it returns the first non-disabled option %#', (options, expected) => {
+    expect(getFirstNonDisabledOption(options)).toEqual(expected);
+  });
+  it('returns the first option if all are disabled', () => {
+    const options = [{ value: 'foo', disabled: true }, { value: 'bar', disabled: true }, { value: 'baz', disabled: true }];
+    expect(getFirstNonDisabledOption(options)).toEqual('foo');
+  });
+});
+
+describe('shouldShowSelectAllStatusAlert', () => {
+  test.each([
```

Comment:
also I just noticed this format instead of the `ddt` data parameters and I love it.

Reply:
ME TOOOOO